diff --git a/src/cli/freenas/cli/functions.py b/src/cli/freenas/cli/functions.py
index 43478f5..c6d5ea0 100644
--- a/src/cli/freenas/cli/functions.py
+++ b/src/cli/freenas/cli/functions.py
@@ -51,6 +51,7 @@ operators = {
     '<': operator.lt,
     '>=': operator.ge,
     '<=': operator.le,
+    '%': operator.mod,
     'and': operator.and_,
     'or': operator.or_,
     'not': operator.not_
diff --git a/src/cli/freenas/cli/parser.py b/src/cli/freenas/cli/parser.py
index 1a93f07..c0f8e6c 100644
--- a/src/cli/freenas/cli/parser.py
+++ b/src/cli/freenas/cli/parser.py
@@ -106,7 +106,7 @@ tokens = list(reserved.values()) + [
     'ASSIGN', 'LPAREN', 'RPAREN', 'EQ', 'NE', 'GT', 'GE', 'LT', 'LE',
     'REGEX', 'UP', 'PIPE', 'LIST', 'COMMA', 'INC', 'DEC', 'PLUS', 'MINUS',
     'MUL', 'DIV', 'EOPEN', 'COPEN', 'LBRACE',
-    'RBRACE', 'LBRACKET', 'RBRACKET', 'NEWLINE', 'COLON', 'REDIRECT'
+    'RBRACE', 'LBRACKET', 'RBRACKET', 'NEWLINE', 'COLON', 'REDIRECT', 'MOD'
 ]
 
 
@@ -231,6 +231,7 @@ t_PLUS = r'\+'
 t_MINUS = r'-'
 t_MUL = r'\*'
 t_DIV = r'\/'
+t_MOD = r'\%'
 t_REGEX = r'~='
 t_COMMA = r'\,'
 t_UP = r'\.\.'
@@ -663,6 +664,7 @@ def p_binary_expr(p):
     binary_expr : expr AND expr
     binary_expr : expr OR expr
     binary_expr : expr NOT expr
+    binary_expr : expr MOD expr
     """
     p[0] = BinaryExpr(p[1], p[2], p[3], p=p)
 

